crates/ty_python_semantic/src/types.rs

```diff
@@ -9147,10 +9090,6 @@ impl<'db> UnionType<'db> {
         Self::from_elements(db, self.elements(db).iter().filter(filter_fn))
     }
 
-    pub fn iter(&self, db: &'db dyn Db) -> Iter<'_, Type<'db>> {
-        self.elements(db).iter()
-    }
```

Comment:
I don't think we really need the indirection here, and removing this method allowed Clippy to point out that there was a callsite in `arguments.rs` where a `union.elements(db).to_vec()` call would be more efficient than calling `union.elements(db).iter().copied().collect()`